app/models/merchant.rb

```diff
@@ -18,4 +18,8 @@ def top_items
     .order(revenue: :desc)
     .limit(5)
   end
+
+  def merchant_invoice_finder
+    Invoice.joins(:items).select(:id).where("items.merchant_id = #{self.id}").group(:id)
```

Comment:
TO continue our discussion, this could also be `items.joins(:invoices).select('invoices.*')`